homeassistant/components/starlink/coordinator.py

```diff
@@ -53,3 +54,11 @@ async def _async_update_data(self) -> StarlinkData:
                 return StarlinkData(*status)
             except GrpcError as exc:
                 raise UpdateFailed from exc
+
+    async def reboot_starlink(self):
+        """Reboot the Starlink system tied to this coordinator."""
+        async with async_timeout.timeout(4):
+            try:
+                await self.hass.async_add_executor_job(reboot, self.channel_context)
+            except GrpcError as exc:
+                raise UpdateFailed from exc
```

Comment:
What should this be throwing, if anything? I feel like `UPDATE_FAILED` isn't right

Reply:
Raise `ValueError` on invalid user input and raise `HomeAssistantError` for other failures such as a problem communicating with a device.